src/qibosoq/rfsoc_server.py

```diff
@@ -415,12 +406,11 @@ def initialize(self):
             if gen_ch not in ch_already_declared:
                 ch_already_declared.append(gen_ch)
 
-                if pulse.frequency < self.max_sampling_rate / 2:
-                    self.declare_gen(gen_ch, nqz=1)
-                else:
-                    self.declare_gen(gen_ch, nqz=2)
+                zone = 1 if pulse.frequency < self.max_sampling_rate / 2 else 2
+                self.declare_gen(gen_ch, nqz=zone)
```

Comment:
Talking about this PR only, these 3 lines are repeated also above.

However, I suspect that in general there is much more duplication in this file.
So, it might be worth to forget about these now, and just refactor the whole file in a separate PR.

Reply:
Yes this code could/should be cleaned for sure.
Actually I already started to do it in the ZCU111 PR, but maybe it's better to open a smaller one just for cleaning this version...

Reply:
Separating concerns is always a good thing, when achievable. Make PRs as atomic as possible, this would also simplify the review process :)